File: DSA Practice/LeetCode/409. Longest Palindrome/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        m = {}
        l = 0
        for i in s:
            if i in m.keys():
                m[i] += 1

            else:
                m[i] = 1

        k = list(m.values())

        for i in range(len(k)):
            if k[i]%2 == 0:
                l += k[i]

            else:
                l += k[i] - 1
                k[i] = 1

        k = sorted(k, reverse=True)

        for i in k:
            if i%2 != 0:
                l += i
                break

        return l

File: DSA Practice/LeetCode/409. Longest Palindrome/solution.java
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character, Integer> m = new HashMap<Character, Integer>();
        int l = 0;
        for (char c : s.toCharArray()){
            if (m.containsKey(c)){
                m.put(c, m.get(c)+1);
            } else {
                m.put(c, 1);
            }
        }

        for (Map.Entry a : m.entrySet()){
            int v = (int) a.getValue();
            if (v%2 == 0){
                l += v;
            } else {
                l += v - 1;
                m.put((char) a.getKey(), 1);
            }
        }

        int mOdd = 0;
        for (Map.Entry a : m.entrySet()){
            int v = (int) a.getValue();
            if (v%2 != 0){
                mOdd = Math.max(mOdd, v);
            }
        }

        l += mOdd;

        return l;
    }
}